core/src/main/scala/kafka/zk/KafkaZkClient.scala

```diff
@@ -1250,6 +1260,8 @@ class KafkaZkClient(zooKeeperClient: ZooKeeperClient, isSecure: Boolean) extends
     while (remainingRequests.nonEmpty) {
       val batchResponses = zooKeeperClient.handleRequests(remainingRequests)
 
+      batchResponses.foreach(response => latencyMetric.update(response.metadata.responseTimeMs))
```

Comment:
@junrao This is another option where we just record the time each request takes (ignoring errors). In a `CONNECTIONLOSS` situation, the latency will be very low, but the `SessionState` should show that. I can file a separate JIRA (since a KIP would be required) for a queue time metric (as it would be good to know if requests are taking longer than usual to be sent).